lib/msf/core/exploit/remote/kerberos/service_authenticator/base.rb

```diff
@@ -347,7 +347,7 @@ def s4u2self(credential, options = {})
       value: credential.keyblock.data.value
     )
 
-    etypes = Set.new([ticket.enc_part.etype])
+    etypes = Set.new([credential.keyblock.enctype.value])
     etypes << Rex::Proto::Kerberos::Crypto::Encryption::RC4_HMAC
```

Comment:
Sync'ing up with @cdelafuente-r7 these etypes are required for his default environment of:
- Windows 2019 DC with AD CS
- Windows 2012 primary DC

In his environment, it breaks if RC4 is not part of etypes

```
msf6 auxiliary(admin/kerberos/get_ticket) > run verbose=true rhosts=192.168.144.224 domain=mylab.local username=serviceA password=123456 action=GET_TGS spn=cifs/dc02.mylab.local impersonate=Administrator
[*] Running module against 192.168.144.224

[+] 192.168.144.224:88 - Received a valid TGT-Response
[*] 192.168.144.224:88 - TGT MIT Credential Cache ticket saved to /Users/cdelafuente/.msf4/loot/20230125130738_default_192.168.144.224_mit.kerberos.cca_624192.bin
[*] 192.168.144.224:88 - Getting TGS impersonating Administrator@mylab.local (SPN: cifs/dc02.mylab.local)
[-] Auxiliary aborted due to failure: unknown: Kerberos Error - KDC_ERR_ETYPE_NOSUPP (14) - KDC has no support for encryption type
[*] Auxiliary module execution completed
```

We can't find anything official in the docs for this, but it seems safe to add :+1: